src/dynarec/rv64/dynarec_rv64_660f.c

```diff
@@ -245,29 +303,43 @@ uintptr_t dynarec64_660F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int
                     DEFAULT;
             }
             break;
-        // case 0x3A:  // these are some more SSSE3+ opcodes
-        //     opcode = F8;
-        //     switch(opcode) {
-        //         case 0x0B:
-        //             INST_NAME("ROUNDSD Gx, Ex, Ib");
-        //             nextop = F8;
-        //             GETEXSD(d0, 0);
-        //             GETGXSD_empty(v0);
-        //             u8 = F8;
-        //             if(u8&4) {
-        //                 u8 = sse_setround(dyn, ninst, x4, x2);
-        //                 FCVTLD(x5, d0, RD_DYN);
-        //                 FCVTDL(v0, x5, RD_DYN);
-        //                 x87_restoreround(dyn, ninst, u8);
-        //             } else {
-        //                 FCVTLD(x5, d0, round_round[u8&3]);
-        //                 FCVTDL(v0, x5, round_round[u8&3]);
-        //             }
-        //             break;
-        //         default:
-        //             DEFAULT;
-        //     }
-        //     break;
+        case 0x3A:  // these are some more SSSE3+ opcodes
+            opcode = F8;
+            switch(opcode) {
+                case 0x0B:
+                    INST_NAME("ROUNDSD Gx, Ex, Ib");
+                    nextop = F8;
+                    GETEXSD(d0, 0);
+                    GETGXSD_empty(v0);
+                    d1 = fpu_get_scratch(dyn);
+                    u8 = F8;
+                    FEQD(x2, d0, d0);
+                    BNEZ_MARK(x2);
+                    FADDD(v0, d0, d0);
+                    B_NEXT_nocond;
+                    MARK; // d0 is not nan
+                    FABSD(v0, d0);
+                    MOV64x(x3, 1ULL << __DBL_MANT_DIG__);
+                    FCVTDL(d1, x3, RD_RTZ);
+                    FLTD(x3, v0, d1);
+                    BNEZ_MARK2(x3);
+                    FMVD(v0, d0);
```

Comment:
you should test if v0!=d0

Reply:
You mean something like `if (v0!=d0) FMVD(v0, d0);`?

Reply:
Yes, no need to move the reg if it's same

Reply:
Done